src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSourceProvider.cs

```diff
@@ -859,6 +855,11 @@ public string ActivePackageSourceName
         /// <param name="source"></param>
         public void SaveActivePackageSource(PackageSource source)
         {
+            if (source is null)
+            {
+                throw new ArgumentNullException(nameof(source));
```

Comment:
In this PR, we implemented `throw new ArgumentException(Resources.Argument_Cannot_Be_Null_Or_Empty, nameof(source));` as exemplified in https://github.com/NuGet/NuGet.Client/pull/5516/files#diff-93482a8631efd30f5df833baa903bcf342fb99a2e93cf9a3dea6a672b44d3a61R308. 

I assume the only difference is that the exception message will be localized when obtained from the resources file. Although I am comfortable with the current implementation, feel free to consider this suggestion if you are addressing other feedback on this PR.

Reply:
The BCL already localizes exception messages.